Amazon.IonDotnet/Internals/Text/IonTextWriter.cs

```diff
@@ -218,6 +218,16 @@ public override void WriteDecimal(BigDecimal value)
             {
                 var decimalString = value.ToString();
                 decimalString = decimalString.Replace('d', 'e');
+
+                // Since Ion Decimal allows an exponent integer following '.' (e.g. 2.d-1) while JSON doesn't, we
+                // should make sure JSON down converter doesn't write any decimal number with '.' immediately
+                // followed by 'e'.
+                var index = decimalString.IndexOf('.', 0);
+                if (index != -1 && decimalString[index + 1] == 'e')
```

Comment:
Will `decimalString[index + 1]` throw `IndexOutOfRange` when you try to write `2.`?

If so, you might want to do this:
```suggestion
                var index = decimalString.IndexOf(".e", 0);
                if (index != -1 && decimalString[index + 1] == 'e')
```

Reply:
Good catch, Thanks!